src/audio/buffers/audio_buffer.c

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Szkudlinski <marcin.szkudlinski@intel.com>
+
+#include <stdint.h>
+#include <stddef.h>
+#include <errno.h>
+#include <rtos/alloc.h>
+#include <sof/audio/audio_buffer.h>
+#include <sof/audio/sink_api.h>
+#include <sof/audio/source_api.h>
+#include <sof/audio/sink_source_utils.h>
+
+#if CONFIG_PIPELINE_2_0
+
+int audio_buffer_attach_secondary_buffer(struct sof_audio_buffer *buffer, bool at_input,
+					 struct sof_audio_buffer *secondary_buffer)
+{
+	if (buffer->secondary_buffer_sink || buffer->secondary_buffer_source)
+		return -EINVAL;
+
+	/* secondary buffer must share audio params with the primary buffer */
+	secondary_buffer->audio_stream_params = buffer->audio_stream_params;
+	/* for performance reasons pointers to params are also kept in sink/src structures */
+	secondary_buffer->_sink_api.audio_stream_params = buffer->audio_stream_params;
+	secondary_buffer->_source_api.audio_stream_params = buffer->audio_stream_params;
```

Comment:
this seems to make 3 copies of stream parameters, so now we have 4 of them on the total? This seems really redundant. Is this a temporary situation until buffers are migrated to the new API? TBH I'm a bit confused about the roadmap of the migration, how far we're into it and what is still left to do. @marcinszkudlinski maybe you could make a "feature" to explain the steps made so far, the current configuration (what types of buffers we currently have, what copying is done, etc.) and the roadmap?